lib/Runtime/Library/RegexHelper.cpp

```diff
@@ -1813,17 +1960,24 @@ namespace Js
         }
     }
 
-    Var RegexHelper::RegexSplit(ScriptContext* entryFunctionContext, JavascriptRegExp* regularExpression, JavascriptString* input, CharCount limit, bool noResult, void *const stackAllocationPointer)
+    Var RegexHelper::RegexSplit(ScriptContext* entryFunctionContext, RecyclableObject* thisObj, JavascriptString* input, CharCount limit, bool noResult, void *const stackAllocationPointer)
     {
-        Var result = RegexHelper::RegexSplitImpl(entryFunctionContext, regularExpression, input, limit, noResult, stackAllocationPointer);
+        Var result = RegexHelper::RegexSplitImpl(entryFunctionContext, thisObj, input, limit, noResult, stackAllocationPointer);
         return RegexHelper::CheckCrossContextAndMarshalResult(result, entryFunctionContext);
     }
 
-    int64 RegexHelper::AdvanceStringIndex(JavascriptString* string, int64 index, bool isUnicode)
+    CharCount RegexHelper::AdvanceStringIndex(JavascriptString* string, CharCount index, bool isUnicode)
     {
         // TODO: Change the increment to 2 depending on the "unicode" flag and
         // the code point at "index". The increment is currently constant at 1
         // in order to be compatible with the rest of the RegExp code.
-        return index + 1;
+
+        if (index + 1 < index) // Overflow
+        {
+            return MaxCharCount;
```

Comment:
I think it's better to cap the index at `MaxCharCount` instead. The new value is used for `lastIndex` and the next call to `exec` would result in **no** match, which would be the same as if we don't cap it.